STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/inc/az_ulib_registry_api.h

```diff
@@ -0,0 +1,123 @@
+ *
+ * @param[in]   key                  The az_span key to look for within the registry.
+ * @param[in]   value                The az_span value corresponding to the input key
+ *
+ * @pre         \p key                            shall not be `NULL`.
+ * @pre         \p sizeof(key)                    shall greater than zero.
+ *
+ * @return The #az_result with the result of the registry operations.
+ *      @retval #AZ_OK                        If retrieving a value from the registry was
+ *                                            successful
+ *      @retval #AZ_ERROR_ITEM_NOT_FOUND      If there are no values that correspond to the
+ *                                            given key within the registry. *
+ *      @retval #AZ_ERROR_ARG                 If the parameters passed into`az_ulib_registry_get`
+ *                                            violate requirements for this function.
+ */
+AZ_NODISCARD az_result az_ulib_registry_get(az_span key, az_span* value);
+
+/**
+ * @brief   This function adds an az_span key and an az_span value into the device registry.
+ *
+ * This function goes through the registry ensuring there are no duplicate elements before
+ * adding a new key value pair to the registry.
+ *
+ * @param[in]   key                  The az_span key to add to the registry.
+ * @param[in]   value                The az_span value to add to the registry.
+ *
+ * @pre         \p key                            shall not be `NULL`.
+ * @pre         \p sizeof(key)                    shall greater than zero.
+ * @pre         \p value                          shall not be `NULL`.
+ * @pre         \p sizeof(value)                  shall greater than zero.
+ *
+ * @return The #az_result with the result of the registry operations.
+ *      @retval #AZ_OK                            If adding a value to the registry was successful
+ *      @retval #AZ_ERROR_ULIB_ELEMENT_DUPLICATE  If there is a key within the registry that is the
+ *                                                same as the new key.
+ *      @retval #AZ_ERROR_ULIB_SYSTEM             If the `az_ulib_registry_add` operation failed on
+ *                                                the system level.
+ *      @retval #AZ_ERROR_ULIB_BUSY               If the resources necesary for the
+ *                                                `az_ulib_registry_add` operation are busy.
+ *      @retval #AZ_ERROR_ARG                     If the parameters passed into
+ *                                                `az_ulib_registry_add` violate requirements for
+ *                                                this function.
+ */
+AZ_NODISCARD az_result az_ulib_registry_add(az_span key, az_span value);
+
+AZ_NODISCARD az_result az_ulib_registry_delete(az_span key);
+
+/**
+ * @brief   This function initializes the device registry.
+ *
+ * This function restores the state of the registry after reboot and repopulates the registry editor
+ * with registry control blocks.
+ *
```

Comment:
Please add a note that this function is not thread safe and the other APIs shall only be invoked after this initialization ends.

Reply:
see L125